src/main.rs

```diff
@@ -267,17 +268,77 @@ fn build(config: &Config) {
     };
 }
 
+fn static_file_handler(dest: &str, req: Request, res: Response) {
+    let path = PathBuf::from(format!("{}{}", dest, req.uri));
+
+    let serve_path = if path.is_file() {
+        // try to point the serve path to `path` if it corresponds to a file
+        path.clone()
+    } else {
+        // try to point the serve path into a "index.html" file in the requested
+        // path
+        path.join("index.html")
+    };
+
+    // if the request points to a file and it exists, read and serve it
+    if serve_path.exists() {
+        let mut file = match File::open(serve_path) {
+            Ok(f) => f,
+            Err(e) => {
+                error!("{}", e);
+                std::process::exit(1);
+            }
+        };
+
+        let mut buffer: Vec<u8> = vec![];
+
+        match file.read_to_end(&mut buffer) {
+            Err(e) => {
+                error!("{}", e);
+                std::process::exit(1);
+            },
+            Ok(_) => {},
+        };
+
+        match res.send(&buffer) {
+            Err(e) => {
+                error!("{}", e);
+                std::process::exit(1);
+            },
+            Ok(()) => {},
+        };
+    }
```

Comment:
I'm trying to set the response status to 404 (or [`StatusCode::NotFound`](http://hyper.rs/hyper/v0.9.1/hyper/status/enum.StatusCode.html)) with the `Response::set_status()`, but it appears to not exist, even if a quick search in the repo tells me [it is supposed to](https://github.com/hyperium/hyper/blob/master/examples/server.rs#L116) ([source](https://github.com/hyperium/hyper/blob/ffb499bdf90a7e7ca441e0f6186c0363070a2e5d/src/server/response.rs#L40)). It apparently is also not documented.

For now, I'm just sending a simple 404 page. I'll dig further into it after I finish these changes.